lib/CAN/CANBus.cpp

```diff
@@ -10,16 +10,106 @@
 #include <CAN.h>
 
 #include <CANBus.h>
-#include <ESP32SJA1000.h>
+#include <CarState.h>
 
+extern CarState carState;
 extern CANBus can;
-extern ESP32SJA1000Class CAN;
 
 void onReceiveForwarder(int packetSize) { can.onReceive(packetSize); }
 
-/*void can_task(void *pvParameter){
-  can.task();
-}*/
+CANBus::CANBus() {
+  // init max ages
+  this->max_ages[BMS_BASE_ADDR] = MAXAGE_BMU_HEARTBEAT;
+  this->max_ages[BMS_BASE_ADDR | 0x1] = MAXAGE_CMU_TEMP;     // CMU1
+  this->max_ages[BMS_BASE_ADDR | 0x4] = MAXAGE_CMU_TEMP;     // CMU2
+  this->max_ages[BMS_BASE_ADDR | 0x7] = MAXAGE_CMU_TEMP;     // CMU3
+  this->max_ages[BMS_BASE_ADDR | 0x2] = MAXAGE_CMU_VOLTAGES; // CMU1
+  this->max_ages[BMS_BASE_ADDR | 0x3] = MAXAGE_CMU_VOLTAGES; // CMU1
+  this->max_ages[BMS_BASE_ADDR | 0x5] = MAXAGE_CMU_VOLTAGES; // CMU2
+  this->max_ages[BMS_BASE_ADDR | 0x6] = MAXAGE_CMU_VOLTAGES; // CMU2
+  this->max_ages[BMS_BASE_ADDR | 0x8] = MAXAGE_CMU_VOLTAGES; // CMU3
+  this->max_ages[BMS_BASE_ADDR | 0x9] = MAXAGE_CMU_VOLTAGES; // CMU3
+  this->max_ages[BMS_BASE_ADDR | 0xF4] = MAXAGE_PACK_SOC;
+  this->max_ages[BMS_BASE_ADDR | 0xF5] = MAXAGE_BALANCE_SOC;
+  this->max_ages[BMS_BASE_ADDR | 0xF6] = MAXAGE_CHARGER_CONTROL;
+  this->max_ages[BMS_BASE_ADDR | 0xF7] = MAXAGE_PRECHARGE_STATUS;
```

Comment:
In modern C++ you should be able to omit the `this->` as long as max_ages as long as the variable is defined in the class. Please do so.